vault/diagnose/tls_verification.go

```diff
@@ -38,6 +38,12 @@ func ListenerChecks(ctx context.Context, listeners []*configutil.Listener) ([]st
 	for _, l := range listeners {
 		listenerID := l.Address
 
+		l.Type = strings.ToLower(l.Type)
```

Comment:
Listener is a pointer, so this mutates it for other callers, which isn't nice. Do you mind rolling this into the next line if it's a pattern used elsewhere? I'm not sure if this really is a case-insensitive field. 

Ty!